libs/api/auth/data-access/src/lib/strategies/api-auth-strategy-telegram.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@nestjs/common'
+import { IdentityProvider } from '@prisma/client'
+import { AuthDataValidator, objectToAuthDataMap } from '@telegram-auth/server'
+import { Response } from 'express-serve-static-core'
+import { ApiAuthService } from '../api-auth.service'
+import { ApiAuthRequest } from '../interfaces/api-auth.request'
+import { ApiAuthStrategyService } from './api-auth-strategy.service'
+
+export interface TelegramAuthPayload {
+  id: number
+  first_name: string
+  last_name?: string
+  username?: string
+  photo_url?: string
+  auth_date: number
+  hash: string
+}
+
+@Injectable()
+export class ApiAuthStrategyTelegramService {
```

Comment:
Following the structure of this project, the controllers and resolvers are very light, the services are where the magic happens.